core/src/main/java/org/elasticsearch/index/query/functionscore/exp/ExponentialDecayFunctionParser.java

```diff
@@ -33,7 +34,12 @@
     }
 
     @Override
-    public ExponentialDecayFunctionBuilder getBuilderPrototype() {
-        return PROTOTYPE;
+    protected ExponentialDecayFunctionBuilder createFromBytes(String fieldName, BytesReference functionBytes) {
+        return new ExponentialDecayFunctionBuilder(fieldName, functionBytes);
+    }
+
+    @Override
+    public ExponentialDecayFunctionBuilder read(StreamInput t) throws IOException {
```

Comment:
I am not too sure when I see that the parser reads the builder from the stream... seems like it was conceptually better  before? maybe that part doesn't belong to the parser?